app/models/user.rb

```diff
@@ -1,6 +1,18 @@
 class User < ApplicationRecord
+  GITHUB_USERNAME_PATTERN = /\A(?!\.)(?!_)[a-z0-9_\.]+(?<!\.)(?<!_)\z/
+
   # Include default devise modules. Others available are:
   # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
   devise :database_authenticatable, :registerable,
-         :recoverable, :rememberable, :validatable
+        :recoverable, :rememberable, :validatable
+
+  has_many :contacts, dependent: :destroy
+  has_many :friendships, dependent: :destroy
+  has_many :friends, through: :friendships
+  has_many :user_settings, dependent: :destroy
+  has_many :media, dependent: :destroy
+  has_many :reminders, dependent: :destroy
+
+  validates :username, uniqueness: true, format: { with: GITHUB_USERNAME_PATTERN, message: "Must be a unique username 🌈" }
```

Comment:
The message is still misleading. it could be a uniq username that doesn't match the gh pattern.

Reply:
To what would you change it?

Reply:
This is what chat gpt says
`Can only contain lowercase letters, numbers, underscores, and periods, and cannot begin or end with a period or underscore`

Reply:
🦾